src/odemis/acq/stream/_helper.py

```diff
@@ -168,13 +172,24 @@ def __init__(self, name, detector, dataflow, emitter, scanner=None, sstage=None,
     # Overrides method of LiveStream
     def _onNewData(self, dataflow, data):
         """
-        Called when a new image has arrived from the detector.
-        Usually the dataflow subscribes to onNewData.
+        Called when a new image has arrived from the detector. Usually the dataflow subscribes to onNewData.
+        If there is integrationTime and there are more than 1 image to be summed, integrate one image after another
+        for live display while running a stream.
         :param dataflow: (model.Dataflow) The dataflow.
         :param data: (model.DataArray). The new image that has arrived from the detector.
         """
-        if hasattr(self, "integrationTime"):
-            self.raw = ([data] + self.raw)[:self.integrationCounts.value]
+        if hasattr(self, "integrationTime") and self.integrationCounts.value > 1:
+            self.integration_step += 1
+            # Once the value of integration counts is reached, prepare the raw data to store the
+            # first acquired image for the next ebeam position and update the integration step.
+            if self.integration_step > self.integrationCounts.value:
```

Comment:
You need to handle the fact that `integrationCounts` can change while playing. The simplest way to handle it is to reset everything if the count is different from the image integrator steps. The more fancy way is that when increasing the integration count (compared to before), then the previous accumulated images are kept. However, as it's quite a bit of a corner case, I think it's fine to leave this fancy way as a TODO.